src/basic/verbs.c

```diff
@@ -23,12 +23,40 @@
 #include <stdbool.h>
 #include <stddef.h>
 
+#include "env-util.h"
 #include "log.h"
 #include "macro.h"
 #include "string-util.h"
 #include "verbs.h"
 #include "virt.h"
 
+/* Wraps running_in_chroot() which is used in various places,
+ * but also adds an environment variable check so external processes
+ * can reliably force this on.
+ */
+bool running_in_chroot_or_offline(void) {
+        /* We've had this condition check for a long time which basically
+         * checks for legacy chroot case like Fedora's
+         * "mock", which is used for package builds.  We don't want
+         * to try to start systemd services there, since without --new-chroot
+         * we don't even have systemd running, and even if we did, adding
+         * a concept of background daemons to builds would be an enormous change,
+         * requiring considering things like how the journal output is handled, etc.
+         * And there's really not a use case today for a build talking to a service.
+         */
+        if (running_in_chroot() > 0)
+                return true;
+
+        /* Added to support use cases like rpm-ostree, where from %post
+         * scripts we only want to execute "preset", but not "start"/"restart"
+         * for example.
+         */
+        if (getenv_bool("SYSTEMCTL_OFFLINE") > 0)
```

Comment:
hmm, I don't like that this is an env var called "SYSTEMCTL_xyz" even though it is checked in verbs.c which is used by more tools than just systemctl. maybe call this SYSTEMD_OFFLINE? yeah, it's only used by systemctl, but maybe the same concept might be useful in other contexts too later on, dunno.

I'd prefer if SYSTEMD_OFFLINE would override the effect of running_in_chroot() in both ways however: i.e. make sure that SYSTEMD_OFFLINE=0 can be used to bypass the chroot check entirely in the opposite way (this would strictly speaking be redundant, since we have SYSTEMD_IGNORE_CHROOT already, but that is both more generic and more specific, than this new env var would be fine, hence this should be OK...

The new env var also needs to be added to ENVIRONMENT.md

Reply:
hmm, and I think we should rename the VERB_NOCHROOT flag to VERB_ONLINE_ONLY or so, and call running_in_chroot_or_offline() just running_offline(), so that the chroot check just becomes an implementation detail of it.